pkgs/build-support/neowrap.nix

```diff
@@ -0,0 +1,294 @@
+{ lib
+, symlinkJoin
+, makeWrapper
+, lndir
+, pkgs
+, extraPkgsByOverride ? []
+}:
+
+# A single pkg or a list of pkgs mainly packaged
+pkgList:
+
+let
+  wrapper = {
+    extraPkgs ? [],
+    extraMakeWrapperArgs ? "",
+    # An attribute set that tells the builder how to handle links per file /
+    # directory found in every key of propagateEnv. E.g:
+    /*** example value ***
+    linkByEnv = {
+      # Tells the builder to link the files in every directory that propagates
+      # XDG_DATA_DIRS.
+      XDG_DATA_DIRS = "link";
+      # Tells the builder to simply link all the files of a package that
+      # propagets this env var. Note that this may affect closure size of the
+      # result as every pkg's all outputs are used unconditionally. Useful for
+      # python / other languages wrappings
+      GI_TYPELIB_PATH = "linkPkg";
+    },
+    /*** Another example value ***
+    linkByEnv ? {
+      QT_PLUGIN_PATH = "link";
+      QML2_IMPORT_PATH = "link";
+    },
+    */
+    # TODO: Decide what to do when `linkPkg` is used here and the file
+    # nix-support/propagated-build-inputs has different values per package
+    linkByEnv ? {},
+    # If we want the wrapping to also include an environmental variable in
+    # out, we list here for every env var what path to add to the wrapper's
+    # values.
+    /*** example value ***
+    wrapOut ? {
+      XDG_DATA_DIRS = "$out/share";
+      PYTHONPATH = "$out/${pkgs.python3.sitePackages}";
+    },
+    */
+    wrapOut ? {},
+  }:
```

Comment:
I wonder if you consider targetting wrappers in-scope? For example, `fwupd` needs to exclude EFI files from being wrapped; `gjs` only wants to wrap installed tests…

Reply:
Am I to understand you are concerned that `wrapGeneric` can't be instructed to wrap only some executables and not others?

Reply:
Yes.

Reply:
Definitely we need to have that ability and it shouldn't be that hard to implement. I still don't have an idea though as to how this interface should look like. I'll update this PR or the RFC once I'll think about something, suggestions are welcome.